homeassistant/components/esphome/ffmpeg_proxy.py

```diff
@@ -172,6 +171,18 @@ async def prepare(self, request: BaseRequest) -> AbstractStreamWriter | None:
         # Only one conversion process per device is allowed
         self.convert_info.proc = proc
 
+        # Create background task which will be cancelled when home assistant shuts down
+        write_task = self.hass.async_create_background_task(
+            self._write_ffmpeg_data(request, writer, proc), "ESPHome media proxy"
+        )
+        await write_task
```

Comment:
Maybe this task should be tied to the ESPHome config entry instead?

Reply:
I'm fine with leaving this at the HA level in case the device reconnects and tries to resume playback.

Reply:
I don't think a reconnect triggers a config entry reload though?

Reply:
Maybe it's fine to abort the task only on HA shutdown, there's no special reason to abort playback just because we unload the config entry.